parser/ast/misc.go

```diff
@@ -3771,3 +3778,42 @@ func (n *CalibrateResourceStmt) Accept(v Visitor) (Node, bool) {
 	n = newNode.(*CalibrateResourceStmt)
 	return v.Leave(n)
 }
+
+type DynamicCalibrateType int
+
+const (
+	// specific time
+	CalibrateStartTime = iota
+	CalibrateEndTime
+	CalibrateDuration
+)
+
+type DynamicCalibrateResourceOption struct {
+	Tp       DynamicCalibrateType
+	Ts       ExprNode
+	StrValue string
+}
+
+func (n *DynamicCalibrateResourceOption) Restore(ctx *format.RestoreCtx) error {
+	if ctx.Flags.HasSkipPlacementRuleForRestoreFlag() {
```

Comment:
Why depending on this flag?

Reply:
No need. Should also remove it in `ResourceGroupOption`.